website/integrations: general cleanup and updates (#12716)
goauthentik/authentik: website/integrations/services/apache-guacamole/index.mdx

>
> -- https://guacamole.apache.org/

:::note

> This documentation only lists the settings that have been changed from their default values. Please verify your changes carefully to avoid any issues accessing your application.

Reviewer: ```suggestion
This documentation lists only the settings that have been changed from their default values. Be aware that any changes other than those explicitly mentioned in this guide could cause issues accessing your application.
```

Reviewer: Let's move this to the Preparation section, please

- `authentik.company` is the FQDN of the authentik installation.

## authentik configuration

1. From the **authentik Admin interface**, navigate to **Applications** -> **Applications**.
2. Use the [wizard](https://docs.goauthentik.io/docs/add-secure-apps/applications/manage_apps#add-new-applications) to create a new application and a **OAuth2/OpenID provider**. During this process:
    - Note the **Client ID**, **Client Secret**, and **slug** values because they will be required later.
    - Set a `Strict` redirect URI to <kbd>https://<em>guacamole.company</em>/</kbd>. If you have configured [Apache Tomcat](https://tomcat.apache.org/) to run Apache Guacamole in a subfolder, you will need to update this value accordingly.
    - Select any available signing key.

## Apache Guacamole Configuration

It is recommended to create an admin account in Guacamole before configuring Single Sign-On to simplify the process. Create a user in Guacamole using the same username as in authentik and grant them admin permissions. This step is important to avoid losing access to the Guacamole admin settings, as you may need to revert your changes without it.

:::warning

> You can configure Apache Guacamole to use either the `sub` or `preferred_username` as the UID field under `user-name-attribute`. When using `preferred_username` as the user identifier, ensure that the [**Allow users to change username** setting](https://docs.goauthentik.io/docs/sys-mgmt/settings#allow-users-to-change-username) is disabled to prevent authentication issues. The `sub` option uses a unique, stable identifier for the user, while `preferred_username` uses the username configured in authentik.

Reviewer: Is this a brand new section, or did you move it from elsewhere and I missed that?

    - Select any available signing key.

## Apache Guacamole Configuration

It is recommended to create an admin account in Guacamole before configuring Single Sign-On to simplify the process. Create a user in Guacamole using the same username as in authentik and grant them admin permissions. This step is important to avoid losing access to the Guacamole admin settings, as you may need to revert your changes without it.

:::warning
You can configure Apache Guacamole to use either the `sub` or `preferred_username` as the UID field under `user-name-attribute`. When using `preferred_username` as the user identifier, ensure that the [**Allow users to change username** setting](https://docs.goauthentik.io/docs/sys-mgmt/settings#allow-users-to-change-username) is disabled to prevent authentication issues. The `sub` option uses a unique, stable identifier for the user, while `preferred_username` uses the username configured in authentik.
:::

import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";

<Tabs
  defaultValue="docker"
  values={[

>     { label: 'Docker Setup', value: 'docker' },

Reviewer: meh. i find the use of Setup to be gratuitous.

    { label: 'Standalone Setup', value: 'standalone' },
  ]}>
  <TabItem value="docker">

>     Docker containers are often configured through environment variables. The following variables are required:

Reviewer: nice edit

```yaml
OPENID_AUTHORIZATION_ENDPOINT=https://<em>authentik.company</em>/application/o/authorize/
OPENID_CLIENT_ID=<em>Client ID from authentik</em>
OPENID_ISSUER=https://<em>authentik.company</em>/application/o/<em>your-slug</em>/
OPENID_JWKS_ENDPOINT=https://<em>authentik.company</em>/application/o/<em>your-slug</em>/jwks/

> OPENID_REDIRECT_URI=https://<em>guacamole.company</em>/ # This must match the Redirect URI set in authentik (Including trailing slash).

Reviewer: ```suggestion
OPENID_REDIRECT_URI=https://<em>guacamole.company</em>/ # This must match the Redirect URI set in authentik (including trailing slash).
```

OPENID_USERNAME_CLAIM_TYPE=preferred_username
```

  </TabItem>
  <TabItem value="standalone">

>     To set up Apache Guacamole in a standalone environment, you'll need to adjust the settings in the `guacamole.properties` file, usually found in the `/etc/guacamole/` directory. Simply add the following:

Reviewer: ```suggestion
    To set up Apache Guacamole in a standalone environment, you'll need to adjust the settings in the `guacamole.properties` file, usually found in the `/etc/guacamole/` directory. Add the following setings:
```

Reviewer: are they settings? The noun needs to be stated in the sentence,,, what exactly is being added...

    { label: 'Standalone', value: 'standalone' },
  ]}>
  <TabItem value="docker">
Docker containers are typically configured using environment variables. To ensure proper integration, add the following variables to your `.env` file:

<IntegrationsMultilineCodeblock>
    {`
OPENID_AUTHORIZATION_ENDPOINT=https://<em>authentik.company</em>/application/o/authorize/
OPENID_CLIENT_ID=<em>Client ID from authentik</em>
OPENID_ISSUER=https://<em>authentik.company</em>/application/o/<em>your-slug</em>/
OPENID_JWKS_ENDPOINT=https://<em>authentik.company</em>/application/o/<em>your-slug</em>/jwks/

> OPENID_REDIRECT_URI=https://<em>guacamole.company</em>/ # Must match Redirect URI in authentik

Reviewer: Should this be italicized? Is that a variable, or exactly what they need to enter?

> OPENID_USERNAME_CLAIM_TYPE=preferred_username

Reviewer: Why an "=" sign here, instead of colon?

Author: = is used in env files while : is used in docker compose environment section

2. To add the certificate as trusted in `/etc/ssl/certs/ca-certificates.crt`, use the following command:

```shell
update-ca-certificates.sh
```

#### Adding Certificate Authority certificate to `/opt/java/openjkd/jre/lib/security/cacerts`

1. To export the certificate of the Certificate Authority, use the following command on the Certificate Authority host:

```shell
openssl pkcs12 -export -in <CA_certificate>.crt -inkey <CA_certificate>.key -out <CA_certificate>.p12 -passout pass:<password>
```

2. To import the certificate to the `/opt/java/openjdk/jre/lib/security/cacerts` keystore on the Apache Guacamole host, use the following command:

```shell
keytool -importkeystore -srckeystore <CA_certificate>.p12 -srcstoretype PKCS12 -keystore /opt/java/openjdk/jre/lib/security/cacerts -deststorepass <destination_store_password> -nopromt -srcstorepass <password>
```

:::note
More information on the keytool command can be found in the [Oracle documentation.](https://docs.oracle.com/en/java/javase/21/docs/specs/man/keytool.html)
:::

> ## Ressources

Reviewer 2: ```suggestion
## Resources
```
